sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java

```diff
@@ -71,6 +73,37 @@ public TableServiceVersion getServiceVersion() {
         return client.getServiceVersion();
     }
 
+    /**
+     * Generates an account SAS for the Azure Storage account using the specified
+     * {@link TableAccountSasSignatureValues}.
+     *
+     * <p>Note : The client must be authenticated via {@link AzureNamedKeyCredential}.
+     * <p>See {@link TableAccountSasSignatureValues} for more information on how to construct an account SAS.</p>
+     *
+     * @param tableAccountSasSignatureValues {@link TableAccountSasSignatureValues}.
+     *
+     * @return A {@link String} representing the SAS query parameters.
+     */
+    public String generateAccountSas(TableAccountSasSignatureValues tableAccountSasSignatureValues) {
```

Comment:
It is a bit confusing to me that this accepts `TableAccountSasSignatureValues` which `TableClient`/`TableAsyncClient` accepts `TableServiceSasSignatureValues`.

Reply:
I agree. I think we may be able to find a better name. Since we copied the classes from `azure-storage-common` I didn't want to simply name them `AccountSasPermission`, for example, to avoid name collisions. Any thoughts @srnagar?

Reply:
What about keeping `TableAccountSas*` for things used to generate an account-level SAS and using `TableSas*` for generating a table-level SAS? It would match the method names: `generateAccountSas()` for `TableServiceClient`/`TableServiceAsyncClient` and `generateSas()` for `TableClient`/`TableAsyncClient`.